Lib/fontTools/feaLib/builder_test.py

```diff
@@ -361,6 +363,92 @@ def test_lookup_inside_feature_aalt(self):
             "Lookup blocks cannot be placed inside 'aalt' features",
             self.build, "feature aalt {lookup L {} L;} aalt;")
 
+    def test_extensions(self):
+        class ast_BaseClass(ast.MarkClass):
+            def asFea(self, indent=""):
+                return ""
+
+        class ast_BaseClassDefinition(ast.MarkClassDefinition):
+            def asFea(self, indent=""):
+                return ""
+
+        class ast_MarkBasePosStatement(ast.MarkBasePosStatement):
+            def asFea(self, indent=""):
+                if isinstance(self.base, ast.MarkClassName) :
+                    res = ""
+                    for bcd in self.base.markClass.definitions :
+                        if res != "" : res += "\n{}".format(indent)
+                        res += "pos base {} {}".format(bcd.glyphs.asFea(), bcd.anchor.asFea())
+                        for m in self.marks :
+                            res += " mark @{}".format(m.name)
+                        res += ";"
+                else :
+                    res = "pos base {}".format(self.base.asFea())
+                    for a, m in self.marks:
+                        res += " {} mark @{}".format(a.asFea(), m.name)
+                    res += ";"
+                return res
+
+        class testAst(object) :
+            MarkBasePosStatement = ast_MarkBasePosStatement
+            def __getattr__(self, name):
```

Comment:
Yes, turning ast into a class, would make the subclassing much tidier. But it makes the core ast much more messy since you would need classes within classes. I suppose this isn't that much, but then the getattr trick isn't so bad. I'm not expecting many people to subclass the ast. And the test gives a good guidance on how to do that subclassing.